src/pynwb/form/backends/hdf5/h5tools.py

```diff
@@ -112,13 +122,24 @@ def __read_dataset(self, h5obj, name=None):
             "dtype": h5obj.dtype,
             "maxshape": h5obj.maxshape
         }
+
+        for key, val in kwargs['attributes'].items():
+            if isinstance(val, bytes): 
+                kwargs['attributes'][key] = val.decode('UTF-8')
+
+
```

Comment:
This hunk and those like it that do "decode" convert bytes to str in the case of reading a python2-saved nwb file with python3